test/plugin_functional/test_suites/core_plugins/application_leave_confirm.ts

```diff
@@ -20,30 +20,86 @@ const getKibanaUrl = (pathname?: string, search?: string) =>
+        : await PageObjects.common.clickConfirmOnModal();
+    } else {
+      log.debug(`defaultTryTimeout * ${timeMultiplier} is not long enough`);
+      return await ensureModalOpen(
+        defaultTryTimeout,
+        (attempts = attempts > 0 ? attempts - 1 : 0),
+        (timeMultiplier = timeMultiplier < 10 ? timeMultiplier + 1 : 10),
+        action
+      );
+    }
+  };
 
-  // Failing: See https://github.com/elastic/kibana/issues/75963
-  // Failing: See https://github.com/elastic/kibana/issues/166838
-  describe.skip('application using leave confirmation', () => {
+  describe('application using leave confirmation', () => {
+    const defaultTryTimeout = config.get('timeouts.try');
+    const attempts = 5;
     describe('when navigating to another app', () => {
+      const timeMultiplier = 2;
       it('prevents navigation if user click cancel on the confirmation dialog', async () => {
         await PageObjects.common.navigateToApp('appleave1');
-        await appsMenu.clickLink('AppLeave 2');
+        await waitForUrlToBe('/app/appleave1');
 
-        await testSubjects.existOrFail('appLeaveConfirmModal');
-        await PageObjects.common.clickCancelOnModal(false);
-        expect(await browser.getCurrentUrl()).to.eql(getKibanaUrl('/app/appleave1'));
+        await appsMenu.clickLink('AppLeave 2');
+        await ensureModalOpen(defaultTryTimeout, attempts, timeMultiplier, 'cancel');
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        await retry.waitFor('navigate to appleave1', async () => {
+          const currentUrl = await browser.getCurrentUrl();
+          log.debug(`currentUrl ${currentUrl}`);
+          return currentUrl.includes('appleave1');
+        });
+        const currentUrl = await browser.getCurrentUrl();
+        expect(currentUrl).to.contain('appleave1');
+        await PageObjects.common.navigateToApp('home');
       });
+
       it('allows navigation if user click confirm on the confirmation dialog', async () => {
         await PageObjects.common.navigateToApp('appleave1');
-        await appsMenu.clickLink('AppLeave 2');
+        await waitForUrlToBe('/app/appleave1');
 
-        await testSubjects.existOrFail('appLeaveConfirmModal');
-        await PageObjects.common.clickConfirmOnModal();
-        expect(await browser.getCurrentUrl()).to.eql(getKibanaUrl('/app/appleave2'));
+        await appsMenu.clickLink('AppLeave 2');
+        await ensureModalOpen(defaultTryTimeout, attempts, timeMultiplier, 'confirm');
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        await retry.waitFor('navigate to appleave1', async () => {
+          const currentUrl = await browser.getCurrentUrl();
+          log.debug(`currentUrl ${currentUrl}`);
+          return currentUrl.includes('appleave2');
+        });
+        const currentUrl = await browser.getCurrentUrl();
+        expect(currentUrl).to.contain('appleave2');
+        await PageObjects.common.navigateToApp('home');
```

Comment:
Navigating away completely is the only way to ensure the browser doesn't get "stuck" in a rendering frenzy.

Reply:
yikes 😄 , do you know what the source is of the rendering frenzy?

Reply:
It's the side nav, the react hooks and the multiple ways navigation can occur. Mostly, though, it's the browser + delays in fading in and out.